pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -1716,8 +1716,54 @@ def clear(self):
         self.curve.clear()
         self.scatter.clear()
 
-    def appendData(self, *args, **kwargs):
-        pass
+
+    def appendData(self, new_x, new_y):
+        """
+        Append new data to the existing dataset.
+
+        .. warning::
+        This method may be significantly slower than calling `setData`, as it 
+        appends to the current data rather than replacing it. For high-performance
+        applications, consider using `setData` when possible.
+
+        .. warning::
+            This method will be undergoing further changes in future versions. 
+            Currently, the accepted arguments are only a subset of those for `setData`.
+            Ensure compatibility by using only the supported arguments.
+    
+        
+        Parameters
+        ----------
+        new_x : array-like
+            The new x-values to append.
+        new_y : array-like
+            The new y-values to append.
+        
+        """
+        if self._dataset is None:
+            # If there is no existing dataset, set the new data as the dataset.
+            self.setData(new_x, new_y)
+            return
+        
+        # Convert new data to arrays
+        new_x = np.array(new_x)
+        new_y = np.array(new_y)
+
+        # Concatenate new data to existing dataset
+        xData = np.concatenate((self._dataset.x, new_x))
+        yData = np.concatenate((self._dataset.y, new_y))
+
+            # Update the internal dataset with the concatenated arrays
+        self._dataset.x = xData
+        self._dataset.y = yData
+        # Invalidate any display data cache
+
+        self._datasetMapped  = None
+        self._datasetDisplay = None  # Invalidate display data
+        self._adsLastValue   = 1
+        
+        # Trigger a visual update of the plot without recalculating everything
+        self.updateItems(styleUpdate=False)  # Only update items, not recalculate all data
```

Comment:
If we do replicate the code here, I am not so sure about the comments on "without recalculating". There's a lot of code that does need to run on the new data, e.g. log scale conversion and range checking.

To me it seems that the main selling point of `appendData()` is that it is possible to feed in updates from some complex data structure (maybe from pandas?) piece by piece, rather than converting the entire thing all the time.

Reply:
That makes sense! Emphasizing the method's ability to handle incremental updates from complex data structures like pandas DataFrames while avoiding any implication of performance optimizations would create clearer expectations for users.
I have revised the comments accordingly